### DISCUSSION TITLE
Pagination: Add page.url.first, page.url.last to the Page interface

### DISCUSSION THREAD

**Discussion by regisphilibert** [withastro/roadmap#862] [Proposal]
# Summary

Adding two new keys to the `page.url` object of the pagination's [Page](https://docs.astro.build/en/guides/routing/#complete-api-reference) interface. 
`page.url.first` would return the url of the first page.
`page.url.last` would return the url of the last page.

# Background & Motivation

Currently when building your own navigation, one must determine what is the first page url in order to build the list of links pointing to various pages. This url can be `/post/` or `/post/category/art/` or `/post/category/art/1` etc... 

This determination can only be worked from `page.url.current` and involves several checks and conditions so that the proper "first page url" is retrieved and validated. For example this is currently what I need to do in order to build basic `firstUrl` and `lastUrl`.

```javascript
const { page } = Astro.props;

// We identify the last part of the url
const pathParts = page.url.current.split('/')
const pathLastPart = pathParts.slice(-1)

let firstPath = page.url.current
// If the last part is a number (`/blog/cat/art/3`), we simply remove it from the pathname to identify the "first page url".
if(!isNaN(pathLastPart)) {
 firstPath = url.replace(/\/[^/]*$/, '');
}
const prevUrl = page.url.prev
const nextUrl = page.url.next
const firstUrl = firstPath
const lastUrl = `${firstPath}/${page.lastPage}`
```
There might be a simpler alternative which I missed out, but I believe this illustrates the time and efforts that this RFP could save for users.

# Goals

- In a pagination context and on any given page, provide users the url for both first page and last page.

# Example

This would greatly simplify the building of any pagination links. Compare the following with the code shared above.

```javascript
const prevUrl = page.url.prev
const nextUrl = page.url.next
const firstUrl = page.url.first
const lastUrl = page.url.last
```

I feel like the naming of the keys follow the convention in use but any suggestion is welcome.

**Comment by delucis** [comment]
Sounds like a good idea to me!

**Comment by tsawada** [comment]
Hi. I opened a PR for this. https://github.com/withastro/astro/pull/11176
Should I create an issue from this discussion as well?

**Reply by bluwy** [reply]
Since the feature is small, I think creating the PR directly is fine.
